mesonbuild/environment.py

```diff
@@ -651,6 +653,95 @@ def __init__(self, source_dir: str, build_dir: str, options: coredata.SharedCMDO
+            if default_libdir in {'lib', 'lib64'}:
+                default_libdirs = [default_libdir, 'share']
+            else:
+                default_libdirs = [default_libdir, 'lib', 'share']
+            for p in default_prefixes:
+                for l in default_libdirs:
+                    default_system_paths.append(p / l / 'pkgconfig')
+            default_library_paths = [Path('/usr') / l for l in default_libdirs]
+            default_include_paths = [Path('/usr/include')]
+
+        def _paths_from_env(varname: str) -> T.Optional[T.List[Path]]:
+            value = _get_env_var(for_machine, is_cross, varname)
+            if value is None:
+                return None
+            return [Path(p) for p in value.split(os.pathsep)] if value else []
+
+        # Paths where to lookup for system pc files
+        if system_paths is None:
+            system_paths = _paths_from_env('PKG_CONFIG_LIBDIR')
+            if system_paths is None:
+                system_paths = default_system_paths
+        else:
+            system_paths = [Path(p) for p in system_paths]
+
+        # Extra paths where to lookup for pc files
+        if extra_paths is None:
+            extra_paths = _paths_from_env('PKG_CONFIG_PATH')
+            if extra_paths is None:
+                extra_paths = []
+        else:
+            extra_paths = [Path(p) for p in extra_paths]
+
+        # Paths where compiler lookup for headers by default. Used to strip
+        # useless -I cflags
+        system_include_paths = _paths_from_env('PKG_CONFIG_SYSTEM_INCLUDE_PATH')
+        if system_include_paths is None:
+            system_include_paths = default_include_paths
+        for var in ['CPATH', 'C_INCLUDE_PATH', 'CPLUS_INCLUDE_PATH', 'OBJC_INCLUDE_PATH']:
+            system_include_paths += _paths_from_env(var) or []
+
+        # Paths where compiler lookup for libraries by default. Used to strip
+        # useless -L cflags
+        system_library_paths = _paths_from_env('PKG_CONFIG_SYSTEM_LIBRARY_PATH')
+        if system_library_paths is None:
+            system_library_paths = default_library_paths
+        system_library_paths += _paths_from_env('LIBRARY_PATH') or []
+
+        # Path prepended to the ${prefix} variable to make the pc file relocatable.
+        if sysroot_dir is None:
+            sysroot_dir = _get_env_var(for_machine, is_cross, 'PKG_CONFIG_SYSROOT_DIR')
+        sysroot_dir = Path(sysroot_dir) if sysroot_dir else None
+
+        # Map the location of a pc file to a sysroot. This is supported by pkgconf
+        # but not pkg-config and is used to have multiple sysroots.
+        sysroot_map = _get_env_var(for_machine, is_cross, 'PKG_CONFIG_SYSROOT_MAP')
+        if sysroot_map is not None:
+            # 'a:b:c:d' -> {'a': 'b', 'c': 'd'}
+            sysroot_map = [Path(p) for p in sysroot_map.split(os.pathsep)]
+            sysroot_map = dict(zip(sysroot_map[::2], sysroot_map[1::2]))
```

Comment:
This might be clearer if you used `itertools.islice`:
```python
sysroot_map = dict(zip(islice(sysroot_map, 0, None, 2), islice(sysroot_map, 1, None, 2)))
```